i-dot-ai/redbox-copilot
# Feature/document status check

## Context

Documents being uploaded should update their status as it goes through the system.

## Changes proposed in this pull request

- Use polling from JS to get django to poll core for info
- Altered the JS to use the json response
- Added client call for getting file status from core
- Added a view to django to get file_status from core or try use s3 file status if present or return unknown

## Guidance to review

- Go to documents page
- Upload a document
- See the status change in the file list

To test error states
- Change the ID used in the file status endpoint to send the file id instead of the core file id
- See that the file status changes to `unknown`

## Relevant links


## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [x] I have tested any code added or changed
- [ ] I have run integration tests
